docs/sources/clients/promtail/configuration.md

```diff
@@ -1016,6 +1019,57 @@ When Promtail receives GCP logs, various internal labels are made available for
 - `__gcp_resource_type`
 - `__gcp_resource_labels_<NAME>`
 
+### Azure Event Hub
+
+The `azure_event_hub` block configures how Promtail receives Azure Event Hubs messages. Promtail uses an Apache Kafka endpoint on Event Hubs to receive messages. More information can be found [here](https://learn.microsoft.com/en-us/azure/event-hubs/azure-event-hubs-kafka-overview).
+
+[Here](https://learn.microsoft.com/en-us/azure/active-directory/reports-monitoring/tutorial-azure-monitor-stream-logs-to-event-hub) is a tutorial on streaming Azure logs to an Azure event hub.
+
+Apache Kafka endpoint is not available within the `Basic` pricing plan. More information about Event Hubs pricing [here](https://azure.microsoft.com/en-us/pricing/details/event-hubs/). 
+
+```yaml
+# Event Hub namespace host names (Required). Typically, it looks like <your-namespace>.servicebus.windows.net:9093.
+fully_qualified_namespace: <string> | default = ""
+
+# Event Hubs to consume (Required).
+event_hubs:
+    [ - <string> ... ]
+
+# Event Hub ConnectionString for authentication on Azure Cloud (Required).
+connection_string: <string> | default = "range"
+
+# The consumer group id.
+[group_id: <string> | default = "promtail"]
+
+# If Promtail should pass on the timestamp from the incoming message or not.
+# When false Promtail will assign the current timestamp to the log when it was processed
+[use_incoming_timestamp: <bool> | default = false]
+
+# If Promtail should ignore messages that don't match the schema for Azure resource logs.
+# Schema is described here https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema.
+[disallow_custom_messages: <bool> | default = false]
+
+# Labels optionally hold labels to associate with each log line.
+[labels]:
+  [ <labelname>: <labelvalue> ... ]
+```
+
+### Available Labels
+
+When Promtail receives Azure Event Hubs messages, various internal labels are made available for [relabeling](#relabel_configs).
+
+**Available Labels:**
+
+- `__azure_event_hub_category`: The log category of the message when a message is an application log.
+
+The list of labels below are discovered because of consuming Kafka interface of Event Hubs:
```

Comment:
Yes 😃 

We use Kafka integration to interact with Event Hubs. Because of this, all Kafka labels are available for this target too.

Your suggestion describes it well. I will update it.

Thank you! 👍